### PR TITLE
refactor: migrate Nats JetStream consumer to durable one

### PR DESCRIPTION
**Pull request by tabVersion**
Signed-off-by: tabVersion <tabvision@bupt.icu>I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

part 2 of https://github.com/risingwavelabs/risingwave/issues/18876
following https://github.com/risingwavelabs/risingwave/pull/18873

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/connector_common/common.rs [hunk 1]
```diff
@@ -683,6 +685,8 @@ impl NatsCommon {
         let consumer = stream
             .get_or_create_consumer(&name, {
                 config.deliver_policy = deliver_policy;
+                config.durable_name = Some(durable_consumer_name);
+                config.filter_subjects = self.subject.split(',').map(|s| s.to_string()).collect();
                 config
```
Comment (yufansong) [inline src/connector/src/connector_common/common.rs:690]: Doc link: https://docs.rs/async-nats/latest/async_nats/jetstream/consumer/pull/struct.Config.html#structfield.durable_name

For filter_subjects, the doc said this, should we care about it?:
```
filter_subjects:
Available on crate feature server_2_10 only.
```

**Comment by tabVersion** [comment]
# A Rough Perf Test

running on my local machine (M1 pro, 8 cores)

ingesting 1,000,000 messages from a single subject of stream, single parallelism

* ack all
  * `create table s2 (a bytea)  with ( connector = 'nats', server_url = 'localhost:4222', subject = 'test_subject', stream = 'test_stream', connect_mode = 'plain', consumer.ack_policy = 'all', durable_consumer_name='test_consumer_2') format plain encode bytes;`
image

* ack explicit
  * `create table s2 (a bytea)  with ( connector = 'nats', server_url = 'localhost:4222', subject = 'test_subject', stream = 'test_stream', connect_mode = 'plain', consumer.ack_policy = 'explicit', durable_consumer_name='test_consumer_3') format plain encode bytes;`
image

* ack none
  * `create table s2 (a bytea)  with ( connector = 'nats', server_url = 'localhost:4222', subject = 'test_subject', stream = 'test_stream', connect_mode = 'plain', consumer.ack_policy = 'none', durable_consumer_name='test_consumer_4') format plain encode bytes;`
image

I am reconsidering the case now... 
We need to scale to 10x parallelism to achieve the none-ack throughput.

**Comment by yufansong** [comment]
> I am reconsidering the case now...
> We need to scale to 10x parallelism to achieve the none-ack throughput.


..... why slow at batch ack/ explicit ack policy

File: src/connector/src/connector_common/common.rs (hunk 1 above)
Comment (tabVersion) [reply src/connector/src/connector_common/common.rs:690]: `server_2_10` is included as the default feature. And they do not mention requirement on the server side.

File: src/connector/src/connector_common/common.rs [hunk 2]
```diff
@@ -680,12 +682,19 @@ impl NatsCommon {
...
-                config
-            })
-            .await?;
+        let consumer = if let Ok(consumer) = stream.get_consumer(&name).await {
+            consumer
+        } else {
+            stream
+                .get_or_create_consumer(&name, {
+                    config.deliver_policy = deliver_policy;
+                    config.durable_name = Some(durable_consumer_name);
```
Comment (KeXiangWang) [inline src/connector/src/connector_common/common.rs:691]: Would be better if we can add some comments mentioning `durable` is the elixir here, maybe in the following PRs.

**Pull request merged**